src/handlers/dd.ts

```diff
@@ -9,29 +9,40 @@ import env from '../env.js';
 const handler: Handler = {
   name: 'dd',
   flags: new HandlerFlags(['RUN_ON_INTERACTION', 'RUN_ON_MESSAGE']),
-  async handle(url: ResolvedURL) {
-    const html = await got(
-      new URL(new URL(url.input).pathname, 'https://ddinstagram.com').toString(),
-    ).text();
-    const regex = /(<meta name="twitter:player:stream" content="(?<video>\/videos\/[a-z0-9_-]+\/\d)"\/><meta name="twitter:player:stream:content_type" content="video\/(?<ext>[a-z0-9_-]+)")|(<meta name="twitter:image" content="(?<image>\/images\/[a-z0-9_-]+\/\d)")/i;
-    const result = regex.exec(html);
-    const ext = result?.groups?.ext || 'png';
-    const fileName = `${url.file}.${ext}`;
-    const path = result?.groups?.video || result?.groups?.image;
+  async handle(url: ResolvedURL, context: HandlerContext) {
+    try {
+      const ddUrl = new URL(url.input);
+      ddUrl.hostname = 'ddinstagram.com';
+      const html = await got(ddUrl).text();
 
-    if (!path) throw new Error('No video or image found');
+      const regex = /(<meta name="twitter:player:stream" content="(?<video>\/videos\/[a-z0-9_-]+\/\d)"\/><meta name="twitter:player:stream:content_type" content="video\/(?<ext>[a-z0-9_-]+)")|(<meta name="twitter:image" content="(?<image>\/images\/[a-z0-9_-]+\/\d)")/i;
+      const { groups } = regex.exec(html) ?? {};
+      if (!groups?.video && !groups?.image) {
+        throw new Error('No video or image found');
+      }
 
-    await download(
-      new URL(path, 'https://ddinstagram.com').toString(),
-      result?.groups?.video ? tmpDir : env.DOWNLOAD_DIR,
-      { filename: fileName },
-    );
+      const ext = context.options?.audioOnly
+        ? context.options.audioFormat ?? 'mp3'
+        : groups.ext ?? 'png';
+      const fileName = `${url.file}.${ext}`;
 
-    if (result?.groups?.video) {
-      await transcode(fileName);
-    }
+      if (!context.fileExists) {
+        const mediaUrl = new URL(groups.video ?? groups.image, 'https://ddinstagram.com');
+        await download(
+          mediaUrl.toString(),
+          groups.video ? tmpDir : env.DOWNLOAD_DIR,
+          { filename: fileName }
+        );
+
+        if (groups.video) {
```

Comment:
If the files exists it has already been transcoded, but that doesn't matter, see above.